Code/GraphMol/SubstructLibrary/Wrap/SubstructLibraryWrap.cpp

```diff
@@ -0,0 +1,213 @@
+const char * SubstructLibraryDoc = "";
+
+struct substructlibrary_wrapper {
+    static void wrap() {
+      // n.b. there can only be one of these in all wrappings
+      //python::class_<std::vector<unsigned int> >("UIntVect").def(
+      //  python::vector_indexing_suite<std::vector<unsigned int>, true>());
+
+      python::class_<MolHolderBase, boost::noncopyable>(
+          "MolHolderBase", "", python::no_init)
+          
+          .def("AddMol", &MolHolderBase::addMol,
+               "Adds molecle to the molecule holder")
+          .def("GetMol", &MolHolderBase::getMol,
+               "Returns a particular molecule in the molecule holder\n\n"
+               "  ARGUMENTS:\n"
+               "    - idx: which molecule to return\n\n"
+               "  NOTE: molecule indices start at 0\n")
+          .def("__len__", &MolHolderBase::size);
+
+      python::class_<MolHolder, boost::shared_ptr<MolHolder>, python::bases<MolHolderBase> >(
+          "MolHolder", MolHolderDoc, python::init<>());
+
+      python::class_<CachedMolHolder, boost::shared_ptr<CachedMolHolder>, python::bases<MolHolderBase > >(
+          "CachedMolHolder", CachedMolHolderDoc, python::init<>())
+          .def("AddBinary", &CachedMolHolder::addBinary,
+               (python::args("pickle")),
+               "Add a binary pickle to the molecule holder, no checking is done on the input data");
+          
+
+      python::class_<CachedSmilesMolHolder, boost::shared_ptr<CachedSmilesMolHolder>, python::bases<MolHolderBase > >(
+          "CachedSmilesMolHolder", CachedSmilesMolHolderDoc, python::init<>())
+          .def("AddTrustedSmiles", &CachedSmilesMolHolder::addTrustedSmiles,
+               (python::args("smiles")),
+               "Add a trusted smiles string to the molecule holder, no checking is done on the input data");
+               
+
+      python::class_<FPHolderBase, boost::shared_ptr<FPHolderBase>, boost::noncopyable>(
+          "FPHolderBase", "", python::no_init)
+          .def("AddMol", &FPHolderBase::addMol,
+               "Adds molecle to the fingerprint database")
+          .def("PassesFilter", &FPHolderBase::passesFilter);
+      
+      python::class_<PatternHolder, boost::shared_ptr<PatternHolder>, python::bases<FPHolderBase> >(
+          "PatternHolder", PatternHolderDoc, python::init<>());
+      
+      python::class_<SubstructLibrary, SubstructLibrary *,
+                     const SubstructLibrary *>(
+                         "SubstructLibrary", SubstructLibraryDoc, python::init<>())
+          .def(python::init<boost::shared_ptr<MolHolderBase> >())          
+          .def(python::init<boost::shared_ptr<MolHolderBase>, boost::shared_ptr<FPHolderBase> >())
+          
+          .def("AddMol",  &SubstructLibrary::addMol,
+               (python::arg("mol")),
+               "Adds a molecule to the substruct library")
+
+
+          .def("GetMatches", (std::vector<unsigned int> (SubstructLibrary::*)(
```

Comment:
The docstrings here and below aren't really up-to-date, but I don't think this is tragic.